Description: Using python3 -m sphinx to build doc
Author: Thomas Goirand <zigo@debian.org>
Forwarded: not-needed
Last-Update: 2018-09-03

Index: openvswitch/Documentation/automake.mk
===================================================================
--- openvswitch.orig/Documentation/automake.mk
+++ openvswitch/Documentation/automake.mk
@@ -121,7 +121,7 @@ EXTRA_DIST += $(DOC_SOURCE)
 
 # You can set these variables from the command line.
 SPHINXOPTS =
-SPHINXBUILD = sphinx-build
+SPHINXBUILD = python3 -m sphinx
 SPHINXSRCDIR = $(srcdir)/Documentation
 SPHINXBUILDDIR = $(builddir)/Documentation/_build
 
