Index: new/debian/openvswitch-switch.init
===================================================================
--- new.orig/debian/openvswitch-switch.init	2014-09-18 08:09:17.000000000 +0200
+++ new/debian/openvswitch-switch.init	2014-09-18 08:09:23.000000000 +0200
@@ -16,10 +16,12 @@
 #
 ### BEGIN INIT INFO
 # Provides:          openvswitch-switch
-# Required-Start:    $network $named $remote_fs $syslog
-# Required-Stop:     $remote_fs
-# Default-Start:     2 3 4 5
-# Default-Stop:      0 1 6
+# Required-Start:    $local_fs
+# Required-Stop:     $local_fs
+# X-Start-Before:    networking
+# X-Stop-After:      networking
+# Default-Start:     S
+# Default-Stop:      0 6
 # Short-Description: Open vSwitch switch
 # Description:       openvswitch-switch provides the userspace components and utilities for
 #                    the Open vSwitch kernel-based switch.
@@ -38,6 +40,13 @@
     [ -n "${bridges}" ] && $1 --allow=ovs ${bridges}
 }
 
+sendsigs_omit() {
+	OMITDIR=/run/sendsigs.omit.d
+	mkdir -p $OMITDIR
+	ln -sf /var/run/openvswitch/ovs-vswitchd.pid $OMITDIR/ovs-vswitchd.pid
+	ln -sf /var/run/openvswitch/ovsdb-server.pid $OMITDIR/ovsdb-server.pid
+}
+
 load_kmod () {
     ovs_ctl load-kmod || exit $?
 }
@@ -61,6 +70,9 @@
 	    exit 0
 	fi
     fi
+
+    sendsigs_omit
+
     set ovs_ctl ${1-start} --system-id=random
     if test X"$FORCE_COREFILES" != X; then
 	set "$@" --force-corefiles="$FORCE_COREFILES"
