Index: new/debian/openvswitch-switch.openvswitch-nonetwork.service
===================================================================
--- /dev/null
+++ new/debian/openvswitch-switch.openvswitch-nonetwork.service
@@ -0,0 +1,12 @@
+[Unit]
+Description=Open vSwitch Internal Unit
+
+DefaultDependencies=no
+After=local-fs.target systemd-tmpfiles-setup.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+EnvironmentFile=-/etc/sysconfig/openvswitch
+ExecStart=/usr/share/openvswitch/scripts/ovs-ctl start --system-id=random $OPTIONS
+ExecStop=/usr/share/openvswitch/scripts/ovs-ctl stop
Index: new/debian/openvswitch-switch.openvswitch.service
===================================================================
--- /dev/null
+++ new/debian/openvswitch-switch.openvswitch.service
@@ -0,0 +1,13 @@
+[Unit]
+Description=Open vSwitch
+After=syslog.target network.target openvswitch-nonetwork.service
+Requires=openvswitch-nonetwork.service
+
+[Service]
+Type=oneshot
+ExecStart=/bin/true
+ExecStop=/bin/true
+RemainAfterExit=yes
+
+[Install]
+WantedBy=multi-user.target
Index: new/debian/rules
===================================================================
--- new.orig/debian/rules
+++ new/debian/rules
@@ -40,6 +40,7 @@ configure-stamp:
 	dh_testdir
 	test -e configure || ./boot.sh
 	test -d _debian || mkdir _debian
+	rm debian/openvswitch-switch.init
 	echo $(DEB_BUILD_OPTIONS)
 	echo $$CC
 	cd _debian && ( \
@@ -130,8 +131,10 @@ binary-common:
 	dh_installexamples
 	dh_installdebconf
 	dh_installlogrotate
-	dh_installinit -p "openvswitch-switch" -R -- start 20 S . stop 80 0 6 .
-	dh_installinit --remaining-packages -R
+	dh_systemd_enable -p "openvswitch-switch" --name openvswitch
+	dh_systemd_enable -p "openvswitch-switch" --name openvswitch-nonetwork
+	dh_installinit
+	dh_systemd_start -p "openvswitch-switch"  --name openvswitch-nonetwork --restart-after-upgrade
 	dh_installcron
 	dh_installman --language=C
 	dh_link
Index: new/debian/control
===================================================================
--- new.orig/debian/control
+++ new/debian/control
@@ -6,7 +6,8 @@ Uploaders: Ben Pfaff <pfaffben@debian.or
 Build-Depends:
  debhelper (>= 8), autoconf (>= 2.64), automake (>= 1.10) | automake1.10, 
  libssl-dev, bzip2, openssl, graphviz,
- python-all (>= 2.6.6-3~), procps, libtool
+ python-all (>= 2.6.6-3~), procps, libtool,
+ dh-systemd (>= 1.4)
 Standards-Version: 3.9.3
 Homepage: http://openvswitch.org/
 
@@ -48,7 +49,7 @@ Description: Open vSwitch common compone
 Package: openvswitch-switch
 Architecture: linux-any
 Suggests: openvswitch-datapath-module
-Depends: ${shlibs:Depends}, ${misc:Depends}, ${python:Depends}, openvswitch-common (= ${binary:Version}), kmod | module-init-tools, procps, uuid-runtime, netbase, python-argparse
+Depends: ${shlibs:Depends}, ${misc:Depends}, ${python:Depends}, openvswitch-common (= ${binary:Version}), kmod | module-init-tools, procps, uuid-runtime, netbase, python-argparse, systemd
 Description: Open vSwitch switch implementations
  Open vSwitch is a production quality, multilayer, software-based,
  Ethernet virtual switch. It is designed to enable massive network
Index: new/Makefile.in
===================================================================
--- new.orig/Makefile.in
+++ new/Makefile.in
@@ -762,7 +762,7 @@ am__noinst_HEADERS_DIST = BUILD.Windows
 	debian/openvswitch-ipsec.install debian/openvswitch-pki.dirs \
 	debian/openvswitch-pki.postinst debian/openvswitch-pki.postrm \
 	debian/openvswitch-switch.README.Debian \
-	debian/openvswitch-switch.dirs debian/openvswitch-switch.init \
+	debian/openvswitch-switch.dirs \
 	debian/openvswitch-switch.install \
 	debian/openvswitch-switch.logrotate \
 	debian/openvswitch-switch.manpages \
@@ -1180,7 +1180,7 @@ EXTRA_DIST = BUILD.Windows CONTRIBUTING
 	debian/openvswitch-ipsec.install debian/openvswitch-pki.dirs \
 	debian/openvswitch-pki.postinst debian/openvswitch-pki.postrm \
 	debian/openvswitch-switch.README.Debian \
-	debian/openvswitch-switch.dirs debian/openvswitch-switch.init \
+	debian/openvswitch-switch.dirs \
 	debian/openvswitch-switch.install \
 	debian/openvswitch-switch.logrotate \
 	debian/openvswitch-switch.manpages \
Index: new/debian/automake.mk
===================================================================
--- new.orig/debian/automake.mk
+++ new/debian/automake.mk
@@ -26,7 +26,6 @@ EXTRA_DIST += \
 	debian/openvswitch-pki.postrm \
 	debian/openvswitch-switch.README.Debian \
 	debian/openvswitch-switch.dirs \
-	debian/openvswitch-switch.init \
 	debian/openvswitch-switch.install \
 	debian/openvswitch-switch.logrotate \
 	debian/openvswitch-switch.manpages \
