Index: openvswitch-2.7.0/debian/openvswitch-switch.openvswitch-nonetwork.service
===================================================================
--- /dev/null
+++ openvswitch-2.7.0/debian/openvswitch-switch.openvswitch-nonetwork.service
@@ -0,0 +1,12 @@
+[Unit]
+Description=Open vSwitch Internal Unit
+
+DefaultDependencies=no
+After=local-fs.target systemd-tmpfiles-setup.service pvenetcommit.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+EnvironmentFile=-/etc/sysconfig/openvswitch
+ExecStart=/usr/share/openvswitch/scripts/ovs-ctl start --system-id=random $OPTIONS
+ExecStop=/usr/share/openvswitch/scripts/ovs-ctl stop
Index: openvswitch-2.7.0/debian/openvswitch-switch.openvswitch.service
===================================================================
--- /dev/null
+++ openvswitch-2.7.0/debian/openvswitch-switch.openvswitch.service
@@ -0,0 +1,13 @@
+[Unit]
+Description=Open vSwitch
+After=syslog.target network.target openvswitch-nonetwork.service
+Requires=openvswitch-nonetwork.service
+
+[Service]
+Type=oneshot
+ExecStart=/bin/true
+ExecStop=/bin/true
+RemainAfterExit=yes
+
+[Install]
+WantedBy=multi-user.target
Index: openvswitch-2.7.0/debian/rules
===================================================================
--- openvswitch-2.7.0.orig/debian/rules
+++ openvswitch-2.7.0/debian/rules
@@ -56,7 +56,10 @@ override_dh_install-indep:
 	dh_install
 
 override_dh_installinit:
-	dh_installinit -R
+	dh_systemd_enable -p "openvswitch-switch" --name openvswitch
+	dh_systemd_enable -p "openvswitch-switch" --name openvswitch-nonetwork
+	dh_installinit
+	dh_systemd_start -p "openvswitch-switch"  --name openvswitch-nonetwork --restart-after-upgrade
 
 override_dh_strip:
 	dh_strip --dbg-package=openvswitch-dbg
Index: openvswitch-2.7.0/debian/control
===================================================================
--- openvswitch-2.7.0.orig/debian/control
+++ openvswitch-2.7.0/debian/control
@@ -14,6 +14,7 @@ Build-Depends: graphviz,
                openssl,
                procps,
                python-all (>= 2.7),
+               dh-systemd (>= 1.4),
                python-six
 Standards-Version: 3.9.3
 Homepage: http://openvswitch.org/
@@ -51,7 +52,8 @@ Depends: kmod | module-init-tools,
          uuid-runtime,
          ${misc:Depends},
          ${python:Depends},
-         ${shlibs:Depends}
+         ${shlibs:Depends},
+         systemd
 Description: Open vSwitch switch implementations
  Open vSwitch is a production quality, multilayer, software-based,
  Ethernet virtual switch. It is designed to enable massive network
Index: openvswitch-2.7.0/debian/automake.mk
===================================================================
--- openvswitch-2.7.0.orig/debian/automake.mk
+++ openvswitch-2.7.0/debian/automake.mk
@@ -24,7 +24,6 @@ EXTRA_DIST += \
 	debian/openvswitch-pki.postrm \
 	debian/openvswitch-switch.README.Debian \
 	debian/openvswitch-switch.dirs \
-	debian/openvswitch-switch.init \
 	debian/openvswitch-switch.install \
 	debian/openvswitch-switch.logrotate \
 	debian/openvswitch-switch.manpages \
