diff --git a/debian/control b/debian/control
index 49d6f5f..71011e6 100644
--- a/debian/control
+++ b/debian/control
@@ -14,9 +14,6 @@ Build-Depends: graphviz,
                openssl,
                procps,
                python-all (>= 2.7),
-               python-qt4,
-               python-twisted-conch,
-               python-zopeinterface
 Standards-Version: 3.9.3
 Homepage: http://openvswitch.org/
 
@@ -38,22 +35,6 @@ Description: Open vSwitch datapath module source - module-assistant version
  from it using module-assistant or make-kpkg.  README.Debian in this
  package provides further instructions.
 
-Package: openvswitch-datapath-dkms
-Architecture: all
-Depends: dkms (>= 1.95), libc6-dev, make, ${misc:Depends}, ${python:Depends}
-Description: Open vSwitch datapath module source - DKMS version
- Open vSwitch is a production quality, multilayer, software-based,
- Ethernet virtual switch. It is designed to enable massive network
- automation through programmatic extension, while still supporting
- standard management interfaces and protocols (e.g. NetFlow, IPFIX,
- sFlow, SPAN, RSPAN, CLI, LACP, 802.1ag). In addition, it is designed
- to support distribution across multiple physical servers similar to
- VMware's vNetwork distributed vswitch or Cisco's Nexus 1000V.
- .
- This package provides the Open vSwitch datapath module source code
- that is needed by openvswitch-switch.  DKMS can built the kernel
- module from it.
-
 Package: openvswitch-common
 Architecture: linux-any
 Depends: openssl,
