diff --git a/debian/openvswitch-switch.postinst b/debian/openvswitch-switch.postinst
index 2464572..160b37b 100755
--- a/debian/openvswitch-switch.postinst
+++ b/debian/openvswitch-switch.postinst
@@ -33,6 +33,10 @@ case "$1" in
                 fi
             done
 	fi
+
+	if dpkg --compare-versions "$2" le "2.6.0-1"; then
+	    rm -f /etc/init.d/openvswitch-switch
+	fi
         ;;
 
     abort-upgrade|abort-remove|abort-deconfigure)
